ophyd/status.py

```diff
@@ -98,9 +103,9 @@ def _settle_then_run_callbacks(self, success=True):
             self.done = True
             self._settled()
 
-            if self._cb is not None:
-                self._cb()
-                self._cb = None
+            for cb in list(self._callbacks):
```

Comment:
This is locked so you can just loop through them and then clear the callbacks I think.